tools/cloudflow-operator/src/main/scala/cloudflow/operator/action/CloudflowStatus.scala

```diff
@@ -0,0 +1,299 @@
+    val streamlets = Try { status.streamletStatuses }.getOrElse(Nil)
+    val streamletStatus =
+      streamlets
+        .find(_.streamletName == streamletName)
+        .map { streamletStatus => updatePod(streamletStatus)(pod) }
+        .toList
+    val streamletStatusesUpdated =
+      streamlets.filterNot(_.streamletName == streamletName) ++ streamletStatus
+
+    status.copy(streamletStatuses = streamletStatusesUpdated, appStatus = calcAppStatus(streamletStatusesUpdated))
+  }
+
+  def deletePod(status: App.AppStatus)(streamletName: String, pod: Pod): App.AppStatus = {
+    val streamlets = Try { status.streamletStatuses }.getOrElse(Nil)
+    val streamletStatus =
+      streamlets
+        .find(_.streamletName == streamletName)
+        .map { streamletStatus => deletePod(streamletStatus)(pod) }
+        .toList
+    val streamletStatusesUpdated =
+      streamlets.filterNot(_.streamletName == streamletName) ++ streamletStatus
+    status.copy(streamletStatuses = streamletStatusesUpdated, appStatus = calcAppStatus(streamletStatusesUpdated))
+  }
+
+  private def ignoreOnErrorStatus(oldApp: Option[App.Cr], newApp: App.Cr) = {
+    val _ = newApp
+    oldApp.map(_.getStatus) match {
+      case Some(status) if status.appStatus == Some(Status.Error) => false
+      case _                                                      => true
+    }
+  }
+
+  implicit val adapter =
+    CustomResourceAdapter[App.Cr, App.List](App.customResourceDefinitionContext)
+
+  // TODO: this failure still happen, seldom and not reproduciple ...
+  // maybe I can reproduce and battle harden deploying 2 operators!
+  def statusUpdateAction(app: App.Cr)(retry: Int = 5): Action = {
+    Action.operation[App.Cr, App.List, Try[App.Cr]](
+      { client: KubernetesClient =>
+        client
+          .customResources(App.customResourceDefinitionContext, classOf[App.Cr], classOf[App.List])
+      }, { cr: MixedOperation[App.Cr, App.List, Resource[App.Cr]] =>
+        Try {
+          val current =
+            cr.inNamespace(app.namespace)
+              .withName(app.name)
+
+          Option(current.fromServer().get()) match {
+            case Some(curr) =>
+              curr.setStatus(app.getStatus)
+
+              val newAppWithStatus = App.Cr(spec = null, metadata = curr.getMetadata, status = curr.getStatus)
+              current.updateStatus(newAppWithStatus)
+            case _ =>
+              if (retry > 0) {
+                throw new Exception("Trying to update a CR that has been deleted")
+              } else {
+                log.warn(s"Cannot yet find the CR to update, giving up")
+                App.Cr(spec = null, metadata = null)
+              }
+          }
+        }
+      }, { res =>
+        res match {
+          case Success(_) => Action.noop
+          case Failure(err) if retry > 0 =>
+            log.error(s"Failure updating the CR status retries: $retry", err)
+            Thread.sleep(100)
```

Comment:
ehhhh

Reply:
lol

Reply:
Retry 5 times 100ms in between? Is this some 'grease' for tests that very quickly undeploy/deploy the same CR?

Reply:
Maybe worth to discuss this one separately.